### PR TITLE
Removes recursion, catches Interrupt exception

### PR DESCRIPTION
**Pull request by RabeaGleissner**
@jsuchy @felipesere

This PR includes the following:
- if a user interrupts the application with `ctrl+c` the application exits more gracefully than before
- the application is now running with a loop (no recursion)

### CODE REVIEW

**Comment by jsuchy** [comment]
LGTM

File: lib/game_runner.rb [hunk 1]
```diff
@@ -16,17 +16,16 @@ def initialize(ui, game, player_factory)
...
-  end
-
-  def replay
-    if ui.replay?
-      start
-    else
+    begin
+      loop do
+        game_option = ui.menu(GAME_OPTIONS)
+        player1, player2 = player_factory.create_players(game_option)
```
Comment (felipesere) [inline lib/game_runner.rb:22]: :+1:

File: lib/player_factory.rb [hunk 2]
```diff
@@ -5,19 +5,13 @@ def initialize(ui)
...
-      ComputerPlayer.new(:X)
-    end
-  end
-
-  def create_player2(game_mode)
-    if game_mode == "1" || game_mode == "3"
-      HumanPlayer.new(:O, ui)
+  def create_players(game_mode)
+    if game_mode == "1"
+      [HumanPlayer.new(:X, ui), HumanPlayer.new(:O, ui)]
```
Comment (felipesere) [inline lib/player_factory.rb:10]: :+1:

**Comment by felipesere** [comment]
LGTM2

**Comment by RabeaGleissner** [comment]
Thanks for the feedback @jsuchy and @felipesere !
Will merge now.

**Pull request merged**